src/neuroconv/utils/json_schema.py

```diff
@@ -70,104 +69,93 @@ def get_base_schema(
     return base_schema
 
 
-def get_schema_from_method_signature(method: Callable, exclude: list = None) -> dict:
+def get_schema_from_method_signature(method: Callable, exclude: Optional[List[str]] = None) -> dict:
+    """Deprecated version of `get_json_schema_from_method_signature`."""
+    message = (
+        "The method `get_schema_from_method_signature` is now named `get_json_schema_from_method_signature`."
+        "This method is deprecated and will be removed after January 2025."
+    )
+    warnings.warn(message=message, category=DeprecationWarning, stacklevel=2)
+
+    return get_json_schema_from_method_signature(method=method, exclude=exclude)
+
+
+def get_json_schema_from_method_signature(method: Callable, exclude: Optional[List[str]] = None) -> dict:
     """
-    Take a class method and return a json-schema of the input args.
+    Get the equivalent JSON schema for a signature of a method.
+
+    Also uses `docstring_parser` (NumPy style) to attempt to find descriptions for the arguments.
 
     Parameters
     ----------
-    method: function
-    exclude: list, optional
+    method : callable
+        The method to generate the JSON schema from.
+    exclude : list of str, optional
+        List of arguments to exclude from the schema generation.
+        Always includes 'self' and 'kwargs'.
 
     Returns
     -------
-    dict
-
+    json_schema : dict
+        The JSON schema corresponding to the method signature.
     """
-    if exclude is None:
-        exclude = ["self", "kwargs"]
-    else:
-        exclude = exclude + ["self", "kwargs"]
-    input_schema = get_base_schema()
-    annotation_json_type_map = dict(
-        bool="boolean",
-        str="string",
-        int="number",
-        float="number",
-        dict="object",
-        list="array",
-        tuple="array",
-        FilePathType="string",
-        FolderPathType="string",
+    exclude = exclude or []
+    exclude += ["self", "cls"]
+
+    signature = inspect.signature(obj=method)
+    parameters = signature.parameters
+    additional_properties = False
+    arguments_to_annotations = {}
```

Comment:
`annotation` is a term used by the `inspect` library. We don't get pydantic field definitions until line 126.

Reply:
Yes, I am aware that that is the term used but the dictionary we are building does not have annotations as items. Plus, the dictionary that we are building is called field_definitions in the pydantic function spec as per the link I showed.

Reply:
> Yes, I am aware that that is the term used but the dictionary we are building does not have annotations as items.

[Line 124](arguments_to_annotations.update({argument_name: (annotation, pydantic_default)})): `arguments_to_annotations.update({argument_name: (annotation, pydantic_default)})`

where `annotation` is `parameter.annotation`

So 'annotation' is the classic Pythonic sense of associating a type with an argument via `:`, and 'annotation' is the built-in reference via the `inspect.signature`

Pydantic models have 'fields' which are themselves annotations and can be thought of as extensions beyond conventional Python annotations, such as what is possible via the `pydantic.Field` class

It is, however, a custom class - and our goal here is to coerce annotations to become simple pydantic fields

Some grey area could occur if/when full `pydantic.Field` support is added/used in the library here, for example, as a way of asserting integer ranges such as `number_of_cpus = pydantic.Field(int, ge=1, le=os.cpu_count(), default=1, description="The number of CPUs to use for the process."). But I'd argue you still don't know ahead of time whether or not any given annotation is going to be a 'Field'

Reply:
I am failing to convey what I mean.

> arguments_to_annotations.update({argument_name: (annotation, pydantic_default)}))
Yes, I understand and I saw it. What I said above is that:
1) "the dictionary does not have annotations as its items". Its items are a tuple.
2) The object that we are building is called is called `field_definitions`. Check my highlight from the link that I shared:

![image](https://github.com/user-attachments/assets/11235472-685d-4910-9ba0-c99b70b691dd)

This is a minimal thing, feel free to ignore it if you guys like `arguments_to_annotations` as a variable name. I am just focusing on what we are building instead of how we are building it.